File: backend/models/ExamSession.js
'use strict';
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const ExamSession = sequelize.define('ExamSession', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    student_id: { type: DataTypes.INTEGER, allowNull: false },
    exam_id: { type: DataTypes.INTEGER, allowNull: false },
    identity_verified: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false },
    verification_score: { type: DataTypes.FLOAT, allowNull: true },
    session_status: {
      type: DataTypes.ENUM('active', 'completed', 'locked'),
      allowNull: false,
      defaultValue: 'active'
    },
    warning_count: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
    answers: { type: DataTypes.JSON, allowNull: true },
    started_at: { type: DataTypes.DATE, allowNull: true },
    submitted_at: { type: DataTypes.DATE, allowNull: true }
  }, {
    tableName: 'exam_sessions',
    underscored: true
  });

  return ExamSession;
};

File: backend/app/models/exam_session.py
from datetime import datetime

from app.extensions import db


class ExamSession(db.Model):
    __tablename__ = "exam_sessions"

    session_id = db.Column(db.Integer, primary_key=True)
    student_id = db.Column(db.Integer, db.ForeignKey("users.user_id"), nullable=False)
    exam_id = db.Column(db.Integer, db.ForeignKey("exams.exam_id"), nullable=False)
    started_at = db.Column(db.DateTime, default=datetime.utcnow)
    submitted_at = db.Column(db.DateTime)
    session_status = db.Column(db.String(20), nullable=False, default="pending")
    identity_verified = db.Column(db.Boolean, nullable=False, default=False)
    warning_count = db.Column(db.Integer, nullable=False, default=0)
    score = db.Column(db.Numeric(5, 2))

    __table_args__ = (db.UniqueConstraint("student_id", "exam_id", name="uq_exam_sessions_student_exam"),)
